be/src/exec/short_circuit_hybrid.cpp

```diff
@@ -0,0 +1,197 @@
+        // tablet 1  <---> k1 =1, k2 =3
+        // tablet 2  <---> k1 =1, k2 =4
+        // this prune need happen in fe
+        auto keys_literal_expr = (*_key_literal_exprs)[i].literal_exprs;
+        size_t num_pk_filters = keys_literal_expr.size();
+        // must all columns
+        if (UNLIKELY(num_pk_filters != _tablet_schema->num_key_columns())) {
+            return Status::Corruption("short circuit only support all key predicate");
+        }
+        for (int j = 0; j < num_pk_filters; ++j) {
+            // init expr context
+            std::vector<ExprContext*> expr_ctxs;
+            std::vector<TExpr> key_literal_expr{keys_literal_expr[j]};
+            // prepare
+            RETURN_IF_ERROR(Expr::create_expr_trees(runtime_state()->obj_pool(), key_literal_expr, &expr_ctxs,
+                                                    runtime_state()));
+            RETURN_IF_ERROR(Expr::prepare(expr_ctxs, runtime_state()));
+            RETURN_IF_ERROR(Expr::open(expr_ctxs, runtime_state()));
+            auto& iteral_expr_ctx = expr_ctxs[0];
+            ASSIGN_OR_RETURN(ColumnPtr value, iteral_expr_ctx->root()->evaluate_const(iteral_expr_ctx));
+            // add const column to chunk
+            auto const_column = ColumnHelper::get_data_column(value.get());
+            _key_chunk->get_column_by_index(j)->append(*const_column);
+        }
+    }
+
+    return Status::OK();
+}
+
+Status ShortCircuitHybridScanNode::_process_value_chunk(std::vector<bool>& found) {
+    std::vector<string> value_field_names;
+    auto value_schema =
+            std::make_unique<Schema>(_tablet_schema->schema(), _tablet_schema->schema()->value_field_column_ids());
+
+    for (auto slot_desc : _tuple_desc->slots()) {
+        auto field = value_schema->get_field_by_name(slot_desc->col_name());
+        if (field != nullptr && !field->is_key()) {
+            value_field_names.emplace_back(field->name());
+        }
+    }
+
+    _value_chunk = ChunkHelper::new_chunk(*(value_schema), _num_rows);
+
+    for (int i = 0; i < _tablets.size(); ++i) {
+        LocalTableReaderParams params;
+        params.version = std::stoi(_versions[i]);
+        params.tablet_id = _tablets[i]->get_tablet_info().tablet_id;
+        _table_reader = std::make_shared<TableReader>();
+        RETURN_IF_ERROR(_table_reader->init(params));
+
+        auto current_chunk = ChunkHelper::new_chunk(*(value_schema), _num_rows);
+        std::vector<bool> curent_found;
+        Status status =
+                _table_reader->multi_get(*(_key_chunk.get()), value_field_names, curent_found, *(current_chunk.get()));
+        if (!status.ok()) {
+            // todo retry
+            LOG(WARNING) << "fail to execute multi get: " << status.detailed_message();
+        }
+
+        // merge all tablet result
+        for (int i = 0; i < curent_found.size(); ++i) {
+            if (curent_found[i]) {
+                found[i] = true;
+            }
+        }
```

Comment:
in order to make this logic correct, requires only at most one tablet return found=true for a specific key, this is guaranteed if there is not bug/problem, if it's not the case it will be very hard to debug, so better to add a check `found[i] != true` check failure indicates failure and should log&return error